addons/stock_account/models/product.py

```diff
@@ -749,8 +749,13 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         candidates = stock_moves\
             .sudo()\
             .filtered(lambda m: is_returned == bool(m.origin_returned_move_id and sum(m.stock_valuation_layer_ids.mapped('quantity')) >= 0))\
-            .mapped('stock_valuation_layer_ids')\
-            .sorted()
+            .mapped('stock_valuation_layer_ids')
+
+        if len(candidates) > 1:
+            candidates = candidates.sorted()
```

Comment:
As discussed in person, this is a neat corner case handling (when a move is associated with only 1 stock val layer, to avoid 1 useless query). There is no point at sorting something if you have a 1 record to sort. Good catch.

Ideally this should be patched in the `sorted()` itself, but not in this PR.